### PR TITLE
Update with-apollo for Apollo 2.0

### PR DESCRIPTION
**Pull request by jthegedus**
I was playing around with `next@beta` and `apollo-client@beta` and thought I would update the example while at it.

The guide here https://github.com/apollographql/apollo-client/blob/master/Upgrade.md gives instructions on how to update. Exposing the apolloCache feels odd, but that comes with splitting the `initApollo` into it's own file. I left it like this for consistency with the previous version.

Since it's using @beta deps it would be ideal to wait until these are all @latest before merging for consistency with the other examples and so not to confuse people.

### CODE REVIEW

**Comment by jthegedus** [comment]
Just found a  discussion of ApolloClient v2 and tracking the Apollo Cache separately to the ApolloClient as is done in this example. A solution was proposed to remove this and go back to only tracking the Client. I will update this later this evening, so pls do not merge. :smile: 

`apolloClient.queryManager.dataStore.getCache().extract()`

**Comment by jthegedus** [comment]
Updated. Just waiting for ApolloClientv2 and Nextv4 to come out of Beta.

**Comment by jthegedus** [comment]
New conversation with some Apollo devs about the implementation of `getCache().extract()` https://github.com/apollographql/apollo-client/issues/2259

**Comment by notrab** [comment]
I like this example update. Great work! 🔥

File: examples/with-apollo/lib/initApollo.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
-import { ApolloClient, createNetworkInterface } from 'react-apollo'
+import ApolloClient from 'apollo-client'
+import Link from 'apollo-link-http'
+import Cache from 'apollo-cache-inmemory'
 import fetch from 'isomorphic-fetch'
 
 let apolloClient = null
+let apolloCache = null
```
Comment (pungggi) [inline examples/with-apollo/lib/initApollo.js]: variable is not used

Comment (jthegedus) [reply examples/with-apollo/lib/initApollo.js]: I'll be updating this example again once the work done https://github.com/apollographql/apollo-client/issues/2259 is available. Hopefully, that is either in the full release or the next beta update.

Comment (prateekrastogi) [reply examples/with-apollo/lib/initApollo.js]: @jthegedus Just checked that all the dependencies involving apollo client are released. So, maybe you can update the pull request and ask for merging too.

Comment (jthegedus) [reply examples/with-apollo/lib/initApollo.js]: @prateekrastogi Yep! I intend to update this within the next 24 hrs :smile:

**Comment by jthegedus** [comment]
@pungggi @prateekrastogi This should be good to go now :smile:

**Comment by prateekrastogi** [comment]
@jthegedus Just integrated it in my codebase, it's working perfectly. Thank you.

**Comment by ads1018** [comment]
Brilliant stuff. I noticed there are two open pull requests for updating apollo client. Which one are we going with?

**Comment by jthegedus** [comment]
@ads1018 yeah, there seems to be more discussion now on the other and they're functionally the same. Given it's their first contribution as well I'll just close this one in favour of the other.

**Pull request closed without merging**